### PR TITLE
chore(ci): refacto gpu bench workflows to reduce duplicates

### PR DESCRIPTION
**Pull request by soonum**
Now there is only one entry point to trigger benchmarks manually. This entry point uses a sub-workflow responsible for provisioning and running the benchmarks.
A weekly workflow is also created with all the targets needed.

This also adds the possibility to run throughput benchmarks on-demand.

### CODE REVIEW

**Comment by soonum** [comment]
> Looks generally good, did you test this ? or I'm guessing you can't because of the workflow not in main rule ?

Yes I have to merge it first and then debug.

File: .github/workflows/benchmark_gpu_integer_common.yml [hunk 1]
```diff
@@ -23,14 +53,67 @@ env:
   SLACK_ICON: https://pbs.twimg.com/profile_images/1274014582265298945/OjBKP9kn_400x400.png
   SLACK_USERNAME: ${{ secrets.BOT_USERNAME }}
   SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
-  BENCH_TYPE: latency
+  FAST_BENCH: TRUE
```
[the remark this answers is not in the document]
Comment (soonum) [reply .github/workflows/benchmark_gpu_integer_common.yml:56]: Nope, look at line 223 of this file.

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/benchmark_gpu_integer_common.yml:56]: prefer setting it in the same step maybe ? easier to reason about local values than to go 170 lines back up

[the remark this answers is not in the document]
Comment (soonum) [reply .github/workflows/benchmark_gpu_integer_common.yml:56]: As you wish.

[the remark this answers is not in the document]
Comment (soonum) [reply .github/workflows/benchmark_gpu_integer_common.yml:56]: It just make the step a one-liner in the `run:` section. Setting it in the same step, would require a bash `if then else`

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/benchmark_gpu_integer_common.yml:56]: I don't have a strong opinion except on the fact that having it in the same place will be easier to change/read

**Review by IceTDrinker — APPROVED** [review]

File: .github/workflows/benchmark_gpu_integer_common.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (soonum) [reply .github/workflows/benchmark_gpu_integer_common.yml:56]: Let's keep it the way it is. From my point of view: less Bash scripting, more readability.

**Pull request merged**